### PR TITLE
Remove msgClient global variable for core-data

### LINKED ISSUE #2027: Data: refactor to remove msgClient global variable
Refactor the above-identified service to remove the above-identified global variable (which is defined in the service's init.go).

This work has already been completed in the system management agent.  Use it as a reference.  See https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/router.go and https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/init.go.

You will need to update route creation to explicitly pass the individual required dependencies to the controller functions.  This will bubble down through the implementation; that is, you'll need to pass those dependencies down the call chain to where they're currently referencing the global variable.  Do not pass the dependency injection container into any controller functions.

You can also create any long-lived dependencies in the service's bootstrap and use the dependency injection container to hold them.  This was done in the system management agent; look there for an example.

Add me as the sole reviewer of your PR.

Note this is Geneva work and should not be merged into master prior to Fuji branching.  As there are a series of these issues for this service (each refactoring away a global variable), assign all of this service's related issues to you and use the branch created for this PR as the starting point for your next PR to avoid extensive conflicts. 

Depends on https://github.com/edgexfoundry/edgex-go/issues/2026

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Fix #2027

Remove the msgClient global variable in favor of using the DI container
for passing around an instance during bootstrap. Update functions which
were referring to the global variable to instead accept an instance as
function arguments.

Update tests to work with the changes to the production code.

Signed-off-by: Anthony M. Bonafide <AnthonyMBonafide@gmail.com>

### CODE REVIEW

**Comment by AnthonyMBonafide** [comment]
These changes where verified with passing blackbox tests.

File: internal/core/data/init.go [hunk 1]
```diff
@@ -96,5 +89,13 @@ func BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer star
 		loggingClient.Error(fmt.Sprintf("failed to create messaging client: %s", err.Error()))
 	}
 
+	dic.Update(di.ServiceConstructorMap{
+		dataContainer.MessagingClientName: func(get di.Get) interface{} {
+			return msgClient
+		},
```
Comment (michaelestrin) [inline internal/core/data/init.go]: You're putting a potentially invalid or nil reference for msgClient into the DIC.  Would it be better to fail-fast and stop the service here?

File: internal/core/data/container/messaging.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package container
+
+import (
+	"github.com/edgexfoundry/go-mod-messaging/messaging"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/di"
```
Comment (michaelestrin) [inline internal/core/data/container/messaging.go]: I've tried to be consistent in putting edgex-go imports before the go-mod-* imports.

File: internal/core/data/init.go (hunk 1 above)
Comment (AnthonyMBonafide) [reply internal/core/data/init.go]: Yes, I think failing fast is the best option here.

Comment (michaelestrin) [reply internal/core/data/init.go]: I don't know if it makes sense contextually (or the way the underlying code that returns msgClient is written) -- you might consider implementing multiple retries before failing.  See [here](https://github.com/edgexfoundry/edgex-go/blob/333829e89581c14c68f5ecbcfda36ed5096dfa50/internal/pkg/bootstrap/handlers/database/database.go#L118) for an example that uses the startup timer to determine when to give up.

**Pull request merged**